### DISCUSSION TITLE
Are agents running parallel ?

### DISCUSSION THREAD

**Discussion by Erdemhan** [mesa/mesa#1735] [Q&A] [answered]
Hi , 

I want to use mesa for a simulation but idk if mesa is the right tool for it. Basically i will have a market model and agents that bid on the market with some parameters. But agents and market have to run discrete and parallel. 

I thought a system like agents can write bids to an in-memory database like redis and the market can read bids from db periodically.

I wonder are mesa agents running parallel  or  or like one after another . İf not is there a known way to do it ?

**Comment by rht** [comment, marked as answer]
There is no framework in Mesa for executing agents in parallel in 1 simulation step yet. But usually, it would have been more efficient if you run separate simulations in each CPU core, in a sensitivity analysis sweep. With that, you can use the `batch_run` function, which can be configured to use all the available cores.

**Comment by Erdemhan** [comment]
Thank you for information
